linkerd/failure-accrual/src/test/scala/io/buoyant/linkerd/failureAccrual/ConfigTest.scala

```diff
@@ -0,0 +1,119 @@
+package io.buoyant.linkerd.failureAccrual
+
+import com.twitter.finagle.service.Backoff
+import com.twitter.util.Duration
+import io.buoyant.config.Parser
+import io.buoyant.linkerd.{ConstantBackoffConfig, FailureAccrualConfig, JitteredBackoffConfig}
+import io.buoyant.test.FunSuite
+import org.scalatest.{Matchers, OptionValues, OutcomeOf}
+import com.twitter.conversions.time._
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.prop.{PropertyChecks, TableDrivenPropertyChecks}
+import org.scalacheck.Gen
+import org.junit.runner.RunWith
+
+/**
+ * Created by eliza on 6/27/17.
+ */
+@RunWith(classOf[JUnitRunner])
+class ConfigTest extends FunSuite
+  with Matchers
+  with OptionValues
+  with PropertyChecks
+  with OutcomeOf {
+
+  def parse(yaml: String): FailureAccrualConfig = {
+    val mapper = Parser.objectMapper(yaml, Seq(Seq(
+      new ConsecutiveFailuresInitializer,
+      new NoneInitializer,
+      new SuccessRateInitializer,
+      new SuccessRateWindowedInitializer
+    )))
+    mapper.readValue[FailureAccrualConfig](yaml)
+  }
+
+  val kinds =
+    Table(
+      ("kind"),
+      ("kind: io.l5d.consecutiveFailures"),
+      ("kind: io.l5d.successRate"),
+      ("kind: io.l5d.successRateWindowed"),
+      ("kind: none")
+    )
+
+  val backoffs =
+    Table(
+      ("config", "backoff"),
+      ("  kind: constant\n  ms: 10000", ConstantBackoffConfig(10000)),
+      ("  kind: jittered\n  maxMs: 10000", JitteredBackoffConfig(None, Some(10000))),
+      ("  kind: jittered\n  minMs: 10000", JitteredBackoffConfig(Some(10000), None)),
+      (
+        """|  kind: jittered
+          |  minMs: 300
+          |  maxMs: 400""".stripMargin,
+        JitteredBackoffConfig(Some(300), Some(400))
+      )
+    )
+
+  test("configs parse to the correct kinds") {
+    // TODO: this could look nicer as a table-based property check
+    parse("kind: io.l5d.consecutiveFailures") shouldBe a[ConsecutiveFailuresConfig]
```

Comment:
Personally, I prefer `should` because I think it's more expressive, and produces more readable failure messages, but I'd be happy to rewrite this to use `assert`.